src/main/java/bio/terra/service/dataset/flight/LockDatasetStep.java

```diff
@@ -43,7 +45,13 @@ public LockDatasetStep(DatasetDao datasetDao, UUID datasetId,
     public StepResult doStep(FlightContext context) {
         try {
             if (sharedLock) {
-                datasetDao.lockShared(datasetId, context.getFlightId());
+                try {
+                    datasetDao.lockShared(datasetId, context.getFlightId());
+                } catch (RetryQueryException retryQueryException) {
+                    return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY);
+                } catch (FatalQueryException fatalQueryException) {
+                    return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL);
+                }
```

Comment:
I *think* you don't need the catch for the FatalQueryException because if lockShared throws an exception it will have a failed step result anyway.